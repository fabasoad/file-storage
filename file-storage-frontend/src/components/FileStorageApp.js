import React from 'react';
import AddOption from './AddOption';
import Header from './Header';
import Files from './Files';
import Settings from './Settings';

export default class FileStorageApp extends React.Component {
    state = {
        files: []
    };
    handleDeleteFiles = () => {
        this.setState(() => ({ files: [] }));
    };
    handleDeleteFile = (option) => {
        this.setState((prevState) => ({
            files: prevState.files.filter(x => x !== option)
        }));
    };
    handleAddFile = (option) => {
        if (!option) {
            return 'Enter valid value to add item';
        } else if (this.state.files.indexOf(option) > -1) {
            return 'This file already exists';
        }
        this.setState((prevState) => ({
            files: prevState.files.concat([option])
        }));
    };
    componentDidMount() {
        const settings = new Settings();
        fetch(`${settings.schema()}://${settings.host()}:${settings.port()}/files`)
        .then(res => res.json())
        .then((data) => {
            this.setState(() => ({ files: data.map(f => f.filename) }));
        })
        .catch(console.log)
    }
    componentDidUpdate(prevProps, prevState) {
        if (prevState.files.length !== this.state.files.length) {
            const json = JSON.stringify(this.state.files);
            localStorage.setItem('files', json);
        }
    }
    componentWillUnmount() {
        console.log('componentWillUnmount');
    }
    render() {
        const subtitle = "All your files in one place";

        return (
            <div id="app">
                <Header subtitle={subtitle} />
                <div className="container">
                    <div className="widget">
                        <Files
                            files={this.state.files}
                            handleDeleteFiles={this.handleDeleteFiles}
                            handleDeleteFile={this.handleDeleteFile}
                        />
                        <AddOption handleAddFile={this.handleAddFile} />
                    </div>
                </div>
            </div>
        );
    }
}