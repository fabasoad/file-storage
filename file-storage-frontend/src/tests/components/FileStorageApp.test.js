import React from 'react';
import { shallow } from 'enzyme';
import FileStorageApp from '../../components/FileStorageApp';

test('should render FileStorageApp correctly', () => {
    const fetch = jest.fn(() => Promise.resolve());
    const wrapper = shallow(<FileStorageApp fetch={fetch} />);
    expect(fetch).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
});