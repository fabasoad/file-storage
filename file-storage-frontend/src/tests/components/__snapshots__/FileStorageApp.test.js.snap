// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FileStorageApp correctly 1`] = `
<div
  id="app"
>
  <Header
    subtitle="All your files in one place"
    title="File Storage"
  />
  <div
    className="container"
  >
    <div
      className="widget"
    >
      <Files
        files={Array []}
        handleDeleteFile={[Function]}
        handleDeleteFiles={[Function]}
      />
      <UploadFile
        handleUploadFile={[Function]}
      />
    </div>
  </div>
</div>
`;
